client/client-lib/src/api.rs

```diff
@@ -357,30 +347,31 @@ impl<C: JsonRpcClient> WsFederationApi<C> {
         &self,
         method: &str,
         param: P,
+        strategy: &mut impl QueryStrategy<R>,
     ) -> Result<R> {
         let params = [serde_json::to_value(param).expect("encoding error")];
-        let requests = self
-            .members
-            .iter()
-            .map(|member| Box::pin(member.request(method, &params)));
-
-        let mut error = None;
-        let mut successes = 0;
-        for request in requests {
-            match request.await {
-                Ok(res) => {
-                    successes += 1;
-                    if successes == 2 * self.max_evil + 1 {
-                        return Ok(res);
-                    }
-                }
-                Err(e) => error = Some(e),
-            };
+        let mut futures = FuturesUnordered::new();
+
+        for member in &self.members {
+            futures.push(member.request(method, &params));
         }
 
-        Err(ApiError::RpcError(
-            error.expect("If there was no success there has to be an error"),
-        ))
+        loop {
```

Comment:
Maybe this could be a function, just to give it a name. If can't, please add a comment.